scripts/wad-to-image

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+# wad-to-image - Convert a WAD file to an image file.
+# by Steven Elliott <selliott512@gmail.com>
+#
+# The WAD file is displayed in Yadex, which must be installed, and then
+# "import" is used to capture the image in the Yadex window.
+#
+# The "level-name" argument specifies the Freedoom name for the level, "c1m1",
+# etc. It should match a file in "levels", but without the ".wad" suffix.
+#
+# The "out-dir" argument is where the captured image is stored. It is created
+# if it does not exist.
+#
+# The "wad-path" optional argument is the patch to the WAD file, such as a
+# modified WAD file in the "levels" directory. By default levels in whatever
+# IWAD Yadex has been configured to use is displayed by Yadex.
+#
+# The "out-name" optional argument for the image file created. By default
+# it's the "level-name" argument.
+#
+# Additional arguments may be passed to Yadex via the YADEX_ARGS environment
+# variable. For example, the width and height can be set to 1024x768 by:
+#     YADEX_ARGS="-width 1024 -height 768" wad-to-image ...
+
+# Globals
+
+bname="${0##*/}"
+small_delay=0.1 # Just a bit to avoid spinning in loops.
+
+# Functions
+
+# Clean up anything that would otherwise be left.
+function cleanup()
+{
+    if [[ -n $yadex_pid ]]
+    then
+        kill $yadex_pid
+    fi
+}
+
+# Translate the level name into a traditional Doom name that Yadex understands.
+function translate_name()
+{
+    local level_name="$1"
+
+    local trans_name="${level_name/c/e}"
+    trans_name="${trans_name/dm/map}"
+
+    echo "$trans_name"
+}
+
+# Main
+
+if [[ ($# -lt 2) || ($# -gt 4) ]]
+then
+    echo "Usage: $bname level-name out-dir [wad-path [out-name]]"
+    exit 1
+fi
+level_name="${1,,}" # lower case
+out_dir="$2"
+if [[ $# -ge 3 ]]
+then
+    wad_path="$3"
+else
+    unset wad_path
+fi
+if [[ $# -ge 4 ]]
+then
+    out_name="$4"
+else
+    out_name="$level_name"
+fi
+
+if [[ $level_name =~ ^c ]]
+then
+    game="doom"
+else
+    game="doom2"
+fi
+
+trap cleanup EXIT
+
+if ! mkdir -p "$out_dir"
+then
+    echo "Unable to create directory out-dir \"$out_dir\"." 1>&2
+    exit 1
+fi
+
+trans_name=$(translate_name "$level_name")
+echo "e $trans_name" | yadex -g "$game" $YADEX_ARGS $wad_path > /dev/null &
+yadex_pid=$!
+if [[ -z $yadex_pid ]]
+then
+    echo "Could not get the PID of Yadex." 1>&2
+    exit 1
+fi
+
+unset yadex_win_id
+for ((i=1; i <= 50; i++))
+do
+    sleep $small_delay
+
+    # The "head" is to get the most recent one.
+    yadex_win_id=$(xwininfo -tree -root | grep -w Yadex | head -n 1 | \
```

Comment:
I was hoping it would work on Linux (where I tested it), or the Cygwin environment you recommend for Windows with an X Server and related tools installed (which I have not tested), but I'll take your work for it that it won't work. It's going away in any case.